MoreCyclopsUpgrades/Managers/PowerManager.cs

```diff
@@ -223,229 +233,38 @@ internal void UpdatePowerSpeedRating()
+            if (rechargeSkip < lastKnownSpeedBoosters + extraSkips)
             {
-                // Recycle surplus power back into the batteries that need it
-
-                if (surplusPower % 2 != 0) // Let this be pseudo-random so we aren't always charging the same battery first each time
-                {
-                    this.SolarChargerMk2.RechargeBatteries(ref surplusPower);
-                    this.ThermalChargerMk2.RechargeBatteries(ref surplusPower);
-                }
-                else
-                {
-                    this.ThermalChargerMk2.RechargeBatteries(ref surplusPower);
-                    this.SolarChargerMk2.RechargeBatteries(ref surplusPower);
-                }
+                rechargeSkip++; // Slightly slows down recharging with more speed boosters and higher difficulty
+                return;
             }
-        }
 
-        /// <summary>
-        /// Gets the total available reserve power across all equipment upgrade modules.
-        /// </summary>
-        /// <returns>The <see cref="int"/> value of the total available reserve power.</returns>
-        internal int GetTotalReservePower()
-        {
-            float availableReservePower = 0f;
-            availableReservePower += this.SolarChargerMk2.TotalBatteryCharge;
-            availableReservePower += this.ThermalChargerMk2.TotalBatteryCharge;
-            availableReservePower += this.NuclearCharger.TotalBatteryCharge;
+            rechargeSkip = 0;
 
-            foreach (CyBioReactorMono reactor in this.CyBioReactors)
-                availableReservePower += reactor.Battery._charge;
+            float powerDeficit = Cyclops.powerRelay.GetMaxPower() - Cyclops.powerRelay.GetPower();
 
-            return Mathf.FloorToInt(availableReservePower);
-        }
+            Manager.HUDManager.UpdateTextVisibility();
 
-        /// <summary>
-        /// Gets the amount of available energy provided by the currently available sunlight.
-        /// </summary>
-        /// <returns>The currently available solar energy.</returns>
-        private float GetSolarChargeAmount()
-        {
-            // The code here mostly replicates what the UpdateSolarRecharge() method does from the SeaMoth class.
-            // Consessions were made for the differences between the Seamoth and Cyclops upgrade modules.
-
-            if (DayNightCycle.main == null)
-                return 0f; // Safety check
+            float power = 0f;
+            foreach (ICyclopsCharger charger in PowerChargers)
+                power += charger.ProducePower(powerDeficit);
 
-            // This is 1-to-1 the same way the Seamoth calculates its solar charging rate.
-
-            return SolarChargingFactor *
-                   DayNightCycle.main.GetLocalLightScalar() *
-                   Mathf.Clamp01((MaxSolarDepth + this.Cyclops.transform.position.y) / MaxSolarDepth); // Distance to surfuce
+            ChargeCyclops(power, ref powerDeficit);
```

Comment:
Difficulty mode could apply a power penalty multiplier here.  
Higher difficulties getting slightly less charge.  
Should be very slightly less. No more than 10% difference at worst.